src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/LongRunningOperationHelper.cs

```diff
@@ -101,6 +102,51 @@ internal string WaitOnOperation(OperationResult operationResult)
 
             this.ProgressTrackerObject.UpdateProgress("Complete", 100);
 
+            // Handle async operation status
+            var result = !string.IsNullOrEmpty(operationResult.Value)
+                ? JObject.Parse(operationResult.Value)
+                : new JObject();
+
+            JToken statusToken;
+            if (result.TryGetValue("status", out statusToken))
+            {
+                TerminalProvisioningStates resourceProvisioningState;
+                if (Enum.TryParse(value: statusToken.ToString(), ignoreCase: true, result: out resourceProvisioningState))
+                {
+                    if (resourceProvisioningState == TerminalProvisioningStates.Succeeded ||
+                    resourceProvisioningState == TerminalProvisioningStates.Ready)
+                    {
+                        return this.GetResourcesClient()
+                            .GetResource<JObject>(resourceId, apiVersion, CancellationToken.None)
+                            .Result
+                            .ToString();
+                    } else if (resourceProvisioningState == TerminalProvisioningStates.Failed ||
+                        resourceProvisioningState == TerminalProvisioningStates.Canceled)
+                    {
+                        JToken errorToken;
+                        if (result.TryGetValue("error", out errorToken))
+                        {
+                            var errors = ResourceManagerSdkClient.ParseDetailErrorMessageWithCode(result.ToString());
+                            var errorText = string.Empty;
+                            if (errors != null)
+                            {
+                                errors.ForEach(error => errorText = string.Join(";", string.Format("Code: {0}, Message: {1}", error.Item1, error.Item2)));
```

Comment:
@vivsriaus should the join call include `errorText`?

```cs
errors.ForEach(error => errorText = string.Join(errorText, ";", string.Format("Code: {0}, Message: {1}", error.Item1, error.Item2)));
```

It seems like without it, `errorText` is going to end up only containing the last error's contents (since it's just joining the `;` character with the formatted error string)